_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp

```diff
@@ -945,6 +943,44 @@ mfxStatus VideoDECODEVP9_HW::PrepareInternalSurface(UMC::FrameMemID &mid, mfxFra
     return MFX_ERR_NONE;
 }
 
+
+static mfxStatus CheckFrameInfo(mfxFrameInfo const &currInfo, mfxFrameInfo &info)
+{
+    MFX_SAFE_CALL(CheckFrameInfoCommon(&info, MFX_CODEC_VP9));
+
+    switch (info.FourCC)
+    {
+        case MFX_FOURCC_NV12:
+        case MFX_FOURCC_AYUV:
+#if (MFX_VERSION >= 1027)
+        case MFX_FOURCC_Y410:
+#endif
+            break;
+        case MFX_FOURCC_P010:
+#if (MFX_VERSION >= 1027)
+        case MFX_FOURCC_Y210:
+#endif
+            MFX_CHECK(info.Shift == 1, MFX_ERR_INCOMPATIBLE_VIDEO_PARAM);
```

Comment:
AFAIR, [Shift == 0] is allowed for system memory. Did u check decoding it?

Reply:
No. But the purpose of this commit is only moving checking function into VP9 decoder code. There are no such check in mfx_common_int, so Shift == 0 with system memory wasn't work.